fe/fe-core/src/main/java/com/starrocks/catalog/EsTable.java

```diff
@@ -141,6 +144,12 @@ public EsTable(long id, String name, List<Column> schema,
         validate(properties);
     }
 
+    public EsTable(long id, String name, List<Column> schema,
+                   Map<String, String> properties, PartitionInfo partitionInfo, String catalogName) throws DdlException {
+        this(id, name, schema, properties, partitionInfo);
```

Comment:
I think you could ref to https://github.com/StarRocks/starrocks/pull/19281 to keep the structure.